src/app/_components/admin/DeleteRoomButton.tsx

```diff
@@ -0,0 +1,52 @@
+import { api } from "~/trpc/react";
+import AdminButton from "~/app/_components/admin/Button";;
+import ConfirmationModal from "./ConfirmationModal";
+import { useState } from 'react';
+import { useRouter } from "next/navigation";
+
+interface DeleteButtonProps {
+  roomId: string;
+}
+
+const DeleteRoomButton: React.FC<DeleteButtonProps> = ({ roomId }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+
+  const deleteItem = api.rooms.delete.useMutation({
+    onMutate: () => setIsLoading(true),
+    onSuccess: () => {
+      setIsLoading(false);
+      setIsModalOpen(false);
+      alert('Item deleted successfully');
```

Comment:
use `toast` instead. Should be able to import it and you can use it to display success and fail messages. [Example usage](https://github.com/dan2811/yamaha/blob/a8b9ee378283c2f23f98320147ba2acf4882fa65/src/app/_components/admin/InstrumentPill.tsx#L30)